google/cloud/spanner/internal/connection_impl.cc

```diff
@@ -1017,6 +1017,12 @@ StatusOr<spanner::CommitResult> ConnectionImpl::CommitImpl(
   request.mutable_request_options()->set_priority(
       ProtoRequestPriority(params.options.request_priority()));
 
+  if (params.options.max_batching_delay().has_value()) {
+    *request.mutable_max_batching_delay() =
```

Comment:
There is no `max_batching_delay` field in the published `CommitRequest`.

Reply:
This field is currently hidden by a TRUSTED_TESTER label. My impression was the order in which I was supposed to do things was:

1. Get this PR (and an equivalent one for other client libraries) approved
2. Remove TT label
3. Merge these PRs.

I have changed my version of this codebase to include the newly generated code when the TT label is removed, but do not have permission to add it to a branch on this codebase. Please let me know if I'm doing something wrong.

Reply:
That sequence will not work for C++.  For complicated reasons[^1], the C++ clients can only compile against public protos.  In this case the public version of [CommitRequest](https://github.com/googleapis/googleapis/blob/e8cca30589a3fadf8e93954bd30197ef12a39ff8/google/spanner/v1/spanner.proto#L990) does not have the field you want to use.


[^1]: The C++ client libraries cannot enforce what version of Protobuf our customers will use, and the generated `*.pb.h` and `*.pb.cc` files can only work with the version of Protobuf that generated them. Therefore, these files are not included in the Cloud C++ source, the customer generates these files, and the customer only has access to the public proto files.

Reply:
Hm, maybe I'm missing something. Step (2) in the sequence I listed will make the CommitRequest proto contain this field. Once we have a PR for each client library that the client library team is happy with, I will remove the visibility label internally and then the public version of CommitRequest will contain the max_batching_delay field. At that point we will be able to merge this PR. Is there a different sequence of steps the client library team usually takes?

Reply:
You need at least:

2.5) Update the googleapis SHA in this repository to use a newer version.
2.6) Get the code to actually compile and pass under CI.

>  Is there a different sequence of steps the client library team usually takes?

I usually don't start writing code in C++ until the protos are published.  I think @devbww has created patches for the proto files and then tweaks the build scripts to apply these patches (there are placeholders in [CMake](https://github.com/googleapis/google-cloud-cpp/blob/611d8445058473b6776abd94d6a8b501243fc36a/external/googleapis/CMakeLists.txt#L139-L149) and [Bazel](https://github.com/googleapis/google-cloud-cpp/blob/611d8445058473b6776abd94d6a8b501243fc36a/bazel/google_cloud_cpp_deps.bzl#L151-L158) to support this).  For obvious reasons that work happens in branches that do not get published to PRs or any public location.

Reply:
As @coryan points out, we really need the new protos published publicly, and google-cloud-cpp reconfigured to use the new protos, before worrying too much about actually using those new protos.

That said, we're happy to comment on the state of the PR, but having it in a non-compilable state make things more difficult.